gateway/src/inference/providers/openai.rs

```diff
@@ -96,10 +97,15 @@ impl InferenceProvider for OpenAIProvider {
         ),
         Error,
     > {
+        if OPENAI_O1_MODELS.contains(&self.model_name.as_str()) {
+            return Err(Error::InvalidRequest {
+                message: "o1 models do not support streaming".to_string(),
+            });
+        }
```

Comment:
Maybe we should check if `model_name` starts with `o1` and if so, categorize it as such. The next one is dropping soon and presumably will have the same limitations.